source/validate_decorations.cpp

```diff
@@ -107,8 +107,23 @@ spv_result_t ValidateDecorations(ValidationState_t& vstate) {
                 "BuiltIn, consumed per entry-point. Entry Point id "
              << entry_point << " does not meet this requirement.";
     }
+    // The LinkageAttributes Decoration cannot be applied to functions targeted
+    // by an OpEntryPoint instruction
+    for (auto& decoration : vstate.id_decorations(entry_point)) {
+      if (SpvDecorationLinkageAttributes == decoration.dec_type()) {
+        const char* linkage_name =
```

Comment:
If we end up doing this a lot, then we should make a helper method on Decoration that returns a utils::string_view or at worst const char* for the string starting at a word pointer.  Basically just hide the ugliness.

But as a distributed systems prof said, "You do something once, it's a trick.  You do it twice, it's a technique."